pkg/federated/fedConfig.go

```diff
@@ -21,42 +24,51 @@ type RemoteSearchService struct {
 	TLSKey  string
 }
 
-func getFederationConfig() []RemoteSearchService {
-	result := getFederationConfigFromSecret()
-	// TODO: Cache the federated configuration result to reduce the overhead.
-	return result
+type fedConfigCache struct {
+	lastUpdated time.Time
+	fedConfig   []RemoteSearchService
 }
 
-// Read the secret search-global-token on each managed hub namespace to get the route token and certificates.
-func getFederationConfigFromSecret() []RemoteSearchService {
-	result := []RemoteSearchService{}
-
-	// Add the global-hub (self) first.
-	client := config.KubeClient()
-	secret, err := client.CoreV1().Secrets("open-cluster-management").Get(context.TODO(), "search-global-token", metav1.GetOptions{})
+var cachedFedConfig = fedConfigCache{
+	lastUpdated: time.Time{},
+	fedConfig:   []RemoteSearchService{},
+}
 
-	if err != nil {
-		klog.Errorf("Error getting secret list: %s", err)
+func getFederationConfig(ctx context.Context, request *http.Request) []RemoteSearchService {
+	if cachedFedConfig.lastUpdated.IsZero() || cachedFedConfig.lastUpdated.Add(60*time.Second).Before(time.Now()) {
```

Comment:
Add to config.

Reply:
resolved with commit https://github.com/stolostron/search-v2-api/pull/189/commits/c1c6f8aea9e70c6b531be736bd4f11a669222d98